CMakeLists.txt

```diff
@@ -263,145 +264,159 @@ endif ()
 
 set (FFTW_VERSION 3.3.7)
 
-set (PREC_SUFFIX)
-if (ENABLE_FLOAT)
-  set (FFTW_SINGLE TRUE)
-  set (BENCHFFT_SINGLE TRUE)
-  set (PREC_SUFFIX f)
+if (BUILD_SHARED_LIBS)
+  add_definitions (-DFFTW_DLL)
 endif ()
 
-if (ENABLE_LONG_DOUBLE)
-  set (FFTW_LDOUBLE TRUE)
-  set (BENCHFFT_LDOUBLE TRUE)
-  set (PREC_SUFFIX l)
-endif ()
+function (add_fftw_library PRECISION)
+  if (PRECISION MATCHES "SINGLE|LDOUBLE|QUAD")
+    set (FFTW_${PRECISION} TRUE)
+    set (BENCHFFT_${PRECISION} TRUE)
+    if (PRECISION STREQUAL SINGLE)
+      set (PREC_SUFFIX f)
+    else ()
+      string (SUBSTRING ${PRECISION} 0 1 PREC_SUFFIX)
+      string (TOLOWER ${PREC_SUFFIX} PREC_SUFFIX)
+    endif()
+  endif ()
 
-if (ENABLE_QUAD_PRECISION)
-  set (FFTW_QUAD TRUE)
-  set (BENCHFFT_QUAD TRUE)
-  set (PREC_SUFFIX q)
-endif ()
-set (fftw3_lib fftw3${PREC_SUFFIX})
+  set (fftw3_lib fftw3${PREC_SUFFIX})
 
-configure_file (cmake.config.h.in config.h @ONLY)
-include_directories (${CMAKE_CURRENT_BINARY_DIR})
+  add_library (${fftw3_lib} ${SOURCEFILES})
 
-if (BUILD_SHARED_LIBS)
-  add_definitions (-DFFTW_DLL)
-endif ()
+  configure_file (cmake.config.h.in ${CMAKE_CURRENT_BINARY_DIR}/${fftw3_lib}/config.h @ONLY)
+  target_include_directories (${fftw3_lib} PRIVATE "${CMAKE_CURRENT_BINARY_DIR}/${fftw3_lib}")
```

Comment:
Use `$<BUILD_INTERFACE:...>` here:
* https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html

Reply:
This line is using `PRIVATE`, so `$<BUILD_INTERFACE:...>` won't make a difference here, as `PRIVATE` includes are only ever used during build, right?

Reply:
Yes, okay.